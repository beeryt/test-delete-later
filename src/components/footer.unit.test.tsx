import React from 'react'
import { render } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')

import Footer from './footer'

describe('Footer', () => {
  const defaultMetadata = {
    copyright: 2021
  }

  beforeAll(() => {
    useStaticQuery.mockImplementation(() => {
      return {
        site: {
          siteMetadata: defaultMetadata
        }
      }
    })
  })

  beforeEach(() => jest.clearAllMocks())

  afterAll(() => useStaticQuery.mockReset())

  it('renders correctly', async () => {
    render(<Footer />)
    expect(document.body).toMatchSnapshot()
    expect(document.querySelector('footer')?.innerHTML).toContain(
      defaultMetadata.copyright
    )
  })
})
