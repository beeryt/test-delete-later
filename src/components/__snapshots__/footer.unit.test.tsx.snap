// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<body>
  <div>
    <footer
      class="footer"
    >
      2021
    </footer>
  </div>
</body>
`;
